regexp.c

```diff
@@ -54,30 +45,22 @@ cRegexp::~cRegexp(void)
 void cRegexp::Compile()
 {
   FreeCompiled();
-  const char *error;
-  int erroffset;
-  re = pcre_compile(regexp, modifiers, &error, &erroffset, NULL);
-  if (error) {
-     error("PCRE compile error: %s at offset %i", error, erroffset);
+  PCRE2_UCHAR error[256];
+  PCRE2_SIZE erroffset;
+  int errnumber;
+  re = pcre2_compile((PCRE2_SPTR8)regexp, PCRE2_ZERO_TERMINATED, modifiers, &errnumber, &erroffset, NULL);
+  if (errnumber) {
+     pcre2_get_error_message(errnumber, error, sizeof(error));
+     error("PCRE compile error: %s at offset %li", error, erroffset);
      enabled = false;
      }
   else {
-     sd = pcre_study(re, PCRE_STUDY_JIT_COMPILE, (const char **)&error);
-     if (error) {
-        error("PCRE study error: %s", error);
-        }
-     else {
-        if (cregexp) {
-           cre = pcre_compile(cregexp, cmodifiers, &error, &erroffset, NULL);
-           if (error) {
-              error("PCRE compile error: %s at offset %i", error, erroffset);
-              enabled = false;
-              }
-           else {
-              csd = pcre_study(cre, PCRE_STUDY_JIT_COMPILE, (const char **)&error);
-              if (error)
-                 error("PCRE study error: %s", error);
-              }
+     if (cregexp) {
+        cre = pcre2_compile((PCRE2_SPTR8)cregexp, PCRE2_ZERO_TERMINATED, cmodifiers, &errnumber, &erroffset, NULL);
+        if (errnumber) {
```

Comment:
This for some reason needs to be "if (!cre)" for the code to work.

Reply:
Done